addons/web_editor/static/src/js/editor/add_snippet_dialog.js

```diff
@@ -192,121 +193,166 @@ export class AddSnippetDialog extends Component {
+                    customSnippetWrapEl.classList.add("o_custom_snippet_wrap");
+                    customSnippetWrapEl.appendChild(snippetPreviewWrapEl);
+                    customSnippetWrapEl.appendChild(editCustomSnippetEl);
+                    containerEl = customSnippetWrapEl;
+
+                    this.__onRenameCustomBtnClick = this._onRenameCustomBtnClick.bind(this);
+                    renameBtnEl.addEventListener("click", this.__onRenameCustomBtnClick);
+                    this.__onDeleteCustomBtnClick = this._onDeleteCustomBtnClick.bind(this);
+                    removeBtnEl.addEventListener("click", this.__onDeleteCustomBtnClick);
+                }
+
+                // Will be sorted in the right columns after
+                containerEl.classList.add('invisible');
+                leftColEl.appendChild(containerEl);
+
+                // Await images.
+                const imageEls = snippetPreviewWrapEl.querySelectorAll("img");
+                // TODO: move onceAllImagesLoaded in web_editor and to use it here
+                return Promise.all(Array.from(imageEls).map(imgEl => {
+                    imgEl.setAttribute("loading", "eager");
+                    return new Promise(resolve => {
+                        if (imgEl.complete) {
+                            resolve();
+                        } else {
+                            imgEl.onload = () => resolve();
+                            // If the image could not be loaded, we still want the
+                            // "d-none" class to be removed.
+                            imgEl.onerror = () => resolve();
+                        }
+                    });
+                })).then(() => containerEl);
             }));
 
-            snippetPreviewWrapEl.classList.remove("d-none");
+            // Sort items in the right column based on their size and the
+            // currently computed size of the column. Note that this does not
+            // compute the size of the column after each element addition: this
+            // is wildly inefficient. Instead: compute the sizes, then add them
+            // all in one go into the right column.
+            const leftColElements = [];
+            const rightColElements = [];
+            for (const itemEl of itemEls) {
+                const size = itemEl.getBoundingClientRect().height;
+                if (leftColSize <= rightColSize) {
+                    leftColElements.push(itemEl);
+                    leftColSize += size;
+                } else {
+                    rightColElements.push(itemEl);
+                    rightColSize += size;
+                }
+            }
+            for (const [colEl, colItemEls] of [
+                [leftColEl, leftColElements],
+                [rightColEl, rightColElements],
+            ]) {
+                for (const el of colItemEls) {
+                    colEl.appendChild(el);
+                    el.classList.remove("invisible");
```

Comment:
If I have a custom snippet and search for something like "cu" to make the custom snippet appear, there is a significant delay before the snippets show up compared to before this commit.